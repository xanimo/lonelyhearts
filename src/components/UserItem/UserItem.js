import React, { Component, PropTypes } from 'react';
import newId from '../NewId/newid';
import { titleize } from '../Titleize/Titleize';
import './UserItem.css';


export default class UserItem extends Component {
    constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.id = newId();
  }

  static propTypes = {
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    phone: PropTypes.string.isRequired,
    cell: PropTypes.string.isRequired,
    largePicture: PropTypes.string
  };
  render() {
    const targetId = "#" + this.id;
    let count = 0;
    count = "#" + this.id;
    console.log("#" + this.id);
    const firstName = this.props.firstName;
    const lastName = this.props.lastName;
    const title = titleize(this.props.title);
    const email = this.props.email;
    const age = this.props.age;
    const phone = this.props.phone;
    const cell = this.props.cell;
    const largePicture = this.props.large;
    return (
    <div className="card">
      <img className="photo" src={largePicture} alt="Profile Avatar"/>
      <div className="card-body">
      <h5 className="card-title">{title} {firstName} {lastName}</h5>
        <h4>{age}</h4>
         
      
          <button type="button" className="btn btn-primary" data-toggle="modal" data-target={targetId}>
            Contact
          </button>

          <div className="modal fade" id={this.id} role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
                <div className="modal-dialog" role="document">
                  <div className="modal-content">
                    <div className="modal-header">
                      <h5 className="modal-title" id="exampleModalLabel">Contact {title} {firstName} {lastName}</h5>
                          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                          </button>
                    </div>
                    <div className="modal-body">
                        <div>Phone: {phone} </div>
                        <div>Cell: {cell} </div>
                        <div>Email: {email} </div>
                    </div>
                <div className="modal-footer">
                  <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
                </div>
            </div>
        </div>
      </div>  
      </div>
    </div>
    );
  }
}
