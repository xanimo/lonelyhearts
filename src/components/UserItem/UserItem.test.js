import renderer from 'react-test-renderer';
import React from 'react';
import UserItem from './UserItem';

test('UserItemTest renders correctly', () => {
  const tree = renderer.create(
              <UserItem
          key="A key"
          title="A title"
          firstName="firstname"
          lastName="lastname"
          email="test@example.com"
          age={23}
          phone="415-555-5555"
          cell="415-514-4155"
          large="<img src='pic.jpg' alt='Profile' />"
          nat="US"/> 
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
// import UserItem from './UserItem';
// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-15';
// Enzyme.configure({ adapter: new Adapter() });
// import { shallow } from 'enzyme';

// describe("UserItem", () => {
//   let props;
//   let mountedUserItem;
//   const UserItem = () => {
//     if (!mountedUserItem) {
//       mountedUserItem = shallow(
//           <UserItem {...props} />
//         );
//     }
//     return mountedUserItem;
//   }

//   beforeEach(() => {
//     props = {
//         firstName: undefined,
//         lastName: undefined,
//         title: undefined,
//         email: undefined,
//         age: undefined,
//         phone: undefined,
//         cell: undefined,
//         largePicture: undefined,
//     };
//     mountedUserItem = undefined;
//   });
  // All tests below


// });
