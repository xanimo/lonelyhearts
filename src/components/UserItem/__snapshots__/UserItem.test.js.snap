// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItemTest renders correctly 1`] = `
<div
  className="card"
>
  <img
    alt="Profile Avatar"
    className="photo"
    src="<img src='pic.jpg' alt='Profile' />"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      A Title
       
      firstname
       
      lastname
    </h5>
    <h4>
      23
    </h4>
    <button
      className="btn btn-primary"
      data-target="#id1"
      data-toggle="modal"
      type="button"
    >
      Contact
    </button>
    <div
      aria-hidden="true"
      aria-labelledby="exampleModalLabel"
      className="modal fade"
      id="id1"
      role="dialog"
    >
      <div
        className="modal-dialog"
        role="document"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
              id="exampleModalLabel"
            >
              Contact 
              A Title
               
              firstname
               
              lastname
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <div>
              Phone: 
              415-555-5555
               
            </div>
            <div>
              Cell: 
              415-514-4155
               
            </div>
            <div>
              Email: 
              test@example.com
               
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
